experiments/trustchain/trustchain_mem_db.py

```diff
@@ -0,0 +1,127 @@
+from Tribler.pyipv8.ipv8.attestation.trustchain.block import TrustChainBlock
+
+
+class TrustchainMemoryDatabase(object):
```

Comment:
Why not just use the `u":memory:"` working directory?

Reply:
Valid question. When doing experiments with the market, I found out that even a sqlite memory database is not efficient enough when trading with a very high volume. If I recall correctly, it had quite a high overhead when validating TrustChain blocks and influenced my results (i.e. the horizontal scalability of the market was not there).

This custom memory database is optimised for block validation and quickly looking up blocks, including lookups of linked blocks. It is completely optional to use, which is the reason I have it in Gumby. Also, it is disabled by default so it does not influence the current validation experiments. I'm pretty sure that I will use this database in the near future, probably to stress the the market and/or TrustChain.

Reply:
Alright.